ipaserver/install/service.py

```diff
@@ -377,10 +377,40 @@ def add_cert_to_service(self):
             logger.critical("Could not add certificate to service %s entry: "
                             "%s", self.principal, str(e))
 
-    def import_ca_certs(self, db, ca_is_configured, conn=None):
+    def import_ca_certs_file(self, cafile, ca_is_configured, conn=None):
+        """
+        Import the CA certificates stored in LDAP into a file
+
+        :param cafile: the file to write the CA certificates to
+        :param ca_is_configured: whether IPA is CA-less or not
+        :param conn: an optional LDAP connection to use
+        """
+        if conn is None:
+            conn = api.Backend.ldap2
+
+        ca_certs = None
+        try:
+            ca_certs = certstore.get_ca_certs(
+                conn, self.suffix, self.realm, ca_is_configured)
+        except errors.NotFound:
+            pass
+        else:
+            with open(cafile, 'wb') as fd:
+                for cert, nickname, trusted, ext_key_usage in ca_certs:
```

Comment:
Maybe something like `x509.write_certificate_list([cert for cert, _nick, _trust, _eku in ca_certs], cafile)`?

Even if you don't choose that way, you'll probably need to prepend the variable names with "_". I did not check with pylint but I am pretty much sure it will yell at you for unused variables.

Reply:
Fixed as part of the pylint tests fixes. Note that I'm purposely leaving one failing for now, the ipa-advise change, because I'm not sure why OCSP has to be disabled.